### ISSUE TITLE
Send email to affiliates

### ISSUE THREAD

**Issue by sumobi** [affiliatewp/AffiliateWP#504]
We're going to implement a new emails class and make all the emails editable from the admin (https://github.com/affiliatewp/AffiliateWP/issues/394), what do we think about allowing the site owner to email the affiliates via the admin? Or is this better suited via an add-on?

It would be handy to be able to email affiliates without having to first export them and then import them into MailPoet/MailChimp etc. Because this functionality would exist in the admin, we could provide an email tag for the affiliate's referral link which automatically forms correct URLs specific for the affiliate.

Related helpscout thread: https://secure.helpscout.net/conversation/65240175/4982/

**Comment by pippinsplugins** [comment]
If we do, let's make sure we implement a queue system so it can be more
reliable on sites that have lower email sending capacity

**Comment by sumobi** [comment]
Good point :+1:

**Comment by sumobi** [comment]
Rami has suggested that whatever we build may be used not for emailing affiliates, but as a general newsletter tool used by the site admin. If this is the case it's going to make our life a lot harder. 

One way we've discussed getting around this is to limit it so it can only send emails if the email address belongs to an affiliate.

**Comment by pippinsplugins** [comment]
I agree that this should only support sending to registered affiliates.

**Comment by GaryJones** [comment]
Maybe jumping the gun, but could the recipients be filtered further? i.e. only registered affiliates who:

* have registered in the last x days (or other time period or date)
* have earned more than $0 in the last x days (or other time period)
* have not had any tracked hits
* have owed payment outstanding more / less than $x
* some other segmenting filter

**Comment by sumobi** [comment]
I really like that idea. Will be extremely useful to quickly filter the affiliates in those ways.

**Comment by pippinsplugins** [comment]
:+1:

**Comment by michaelbeil** [comment]
https://github.com/AffiliateWP/AffiliateWP/issues/504

**Comment by michaelbeil** [comment]
https://secure.helpscout.net/conversation/540117629/77984?folderId=205529
